pkg/kube/annotations/annotations.go

```diff
@@ -0,0 +1,84 @@
+package annotations
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Versioned interface {
+	client.Object
+	GetMongoDBVersionForAnnotation() string
+	NamespacedName() types.NamespacedName
+	IsChangingVersion() bool
+}
+
+type patchValue struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+const (
+	LastAppliedMongoDBVersion = "mongodb.com/v1.lastAppliedMongoDBVersion"
+)
+
+func getAnnotation(object Versioned, key string) string {
```

Comment:
Can we make this method public like "SetAnnotations". It will also enable us to get the keys without auxiliary methods like `GetLastAppliedMongoDBVersion`. WDYT?